components/reddit/reddit.app.js

```diff
@@ -0,0 +1,58 @@
+const axios = require("axios");
+const get = require("lodash.get");
+
+module.exports = {
+  type: "app",
+  app: "reddit",
+  methods: {
+    _accessToken() {
+      return this.$auth.oauth_access_token;
+    },
+    _apiUrl() {
+      return `https://oauth.reddit.com`;
+    },
+    async _makeRequest(opts) {
+      if (!opts.headers) opts.headers = {};
+      opts.headers.authorization = `Bearer ${this._accessToken()}`;
+      opts.headers["user-agent"] = "@PipedreamHQ/pipedream v0.1";
+      const { path } = opts;
+      delete opts.path;
+      opts.url = `${this._apiUrl()}${path[0] === "/" ? "" : "/"}${path}`;
+      return (await axios(opts)).data;
+    },
+    wereLinksPulled(reddit_things) {
+      const links_length = get(reddit_things, "data.children.length");
+      return (
+        links_length !== null &&
+        links_length !== undefined &&
+        reddit_things.data.children.length > 0
+      );
+    },
+
+    async getNewHotSubredditPosts(subreddit, g, show, sr_detail) {
+      let params = new Object();
+      if (show == "all") {
+        params["show"] = show;
+      }
+      params["g"] = g;
+      params["sr_detail"] = sr_detail;
+
+      return await await this._makeRequest({
+        path: `/r/${subreddit}/hot`,
+        params: {
+          limit: 10,
```

Comment:
Is there a reason the limit is set to `10` instead of parameterized?

Reply:
The limit is set to 10 is to honor the stated description in the source under the GitHub issue where it was raised.

> New Hot Post in a Subreddit (post in the top 10 of a subreddit

Looking at the no code database, this is taken from Zapier. Reddit API endpoint for new host post can handle a max of 100 per batch so if you prefer it can be parametrized.

One final aspect, pinned post in the subreddit are returned with this endpoint, so if the input subreddit (parametrized) had two pinned posts, even with a limit of 10, the additional 2 pinned would be returned. As didn't programmatically, but we could. Since we are using the dedupe unique strategy, they would be printed only once.

Reply:
I see, thanks. Given the deduper logic, I think it's fine to emit pinned posts, since as you note they'll be ignored by the deduper.

I'd recommend removing the hardcoded limit of 10 here from the method (so that future sources / actions can use it with any limit), and pass the limit of 10 from the source itself.

Reply:
Ok, i set the limit value from the source file